lambdas/data-migration2/src/granulesAndFiles.ts

```diff
@@ -103,22 +110,37 @@ export const migrateGranuleRecord = async (
 
   let existingRecord;
 
-  try {
-    existingRecord = await granulePgModel.get(trx, {
-      granule_id: record.granuleId,
-      collection_cumulus_id: collectionCumulusId,
-    });
-  } catch (error) {
-    if (!(error instanceof RecordDoesNotExist)) {
-      throw error;
+  if (migrateAndOverwrite !== 'true') {
+    let recordCheckError;
+    try {
+      existingRecord = await granulePgModel.get(trx, {
+        granule_id: record.granuleId,
+        collection_cumulus_id: collectionCumulusId,
+      });
+    } catch (error) {
+      if (!(error instanceof RecordDoesNotExist)) {
+        throw error;
+      }
+      recordCheckError = error;
     }
-  }
 
-  const isExistingRecordNewer = existingRecord
-    && existingRecord.updated_at >= new Date(record.updatedAt);
+    if (migrateOnlyFiles === 'true') {
+      if (!existingRecord) {
+        throw recordCheckError;
+      }
+      return existingRecord.cumulus_id;
+    }
+
+    const isExistingRecordNewer = existingRecord
+      && existingRecord.updated_at >= new Date(record.updatedAt);
+
+    if (isExistingRecordNewer) {
+      throw new RecordAlreadyMigrated(`Granule ${record.granuleId} was already migrated, skipping`);
+    }
+  }
 
-  if (isExistingRecordNewer) {
-    throw new RecordAlreadyMigrated(`Granule ${record.granuleId} was already migrated, skipping`);
+  if (migrateOnlyFiles === 'true') {
+    throw new InvalidArgument('Invalid migration parameters detected, migrateOnlyFiles cannot be set to true if migrateAndOverwrite is also set to true');
```

Comment:
Would it be clearer to put this validation at the top of the function and explicitly check 
```
if (migrateOnlyFiles === 'true' && migrateAndOverwrite === 'true) { ... }
```
?

I like doing validation first but either way is totally fine, feel free to ignore.

Reply:
No, this is a catch for if  `if (migrateAndOverwrite !== 'true') {` *and* this case.

We're catching it in the top medthod as well, so this *might* be overkill :thinking:

Reply:
Yeah, I think I just wasn't following the logic here. It looks like we're assuming that `migrateAndOverwrite` is also true when we've gotten to this point. That is valid because we check on line 137 `if (migrateOnlyFiles === 'true')` so it does look like, when you reach line 152, we can assume that `migrateAndOverwrite` is true.

It just took me a minute to see why we can assume `migrateAndOverwrite` is true, you know? That's why I was looking for an explicit, duplicate check somewhere up top. Really just so I don't have to think 😄 

There's no change necessary here. What you have is great.